src/backend/distributed/commands/dependencies.c

```diff
@@ -56,6 +58,12 @@ EnsureDependenciesExistOnAllNodes(const ObjectAddress *target)
 	List *dependenciesWithCommands = NIL;
 	List *ddlCommands = NULL;
 
+	/*
+	 * If there is any unsupported dependency or circular dependency exists, Citus can
+	 * not ensure dependencies will exist on all nodes.
+	 */
+	EnsureDependenciesCanBeDistributed(target);
```

Comment:
Do we need additional EnsureDependenciesExistOnAllNodes for the missing places with this PR?

Reply:
I'm planning to handle it with a separate PR.